src/viem-config.ts

```diff
@@ -1,11 +1,18 @@
 import { defineChain } from 'viem';
 import { Chain } from './types';
 import { CHAINS } from './generated/chains';
+import { toUpperSnakeCase } from './utils/strings';
+
+export function chainHttpRpcUrlName(chain: Chain) {
+  return `VIEM_HTTP_RPC_URL_${toUpperSnakeCase(chain.alias)}`;
```

Comment:
Hardhat uses the `HARDHAT_` prefix which doesn't really transfer here (unless we want to force it) so I ended up going with `VIEM_`, even though the values would almost always be the same. I guess it's quite unlikely someone would have both (?), otherwise something more generic would be better (but also a potentially breaking change).

If we don't want a breaking change, we could also do:

```ts
`HARDHAT_HTTP_RPC_URL_${toUpperSnakeCase(chain.alias)}` || `HTTP_RPC_URL_${toUpperSnakeCase(chain.alias)` || defaultProvider!.rpcUrl!
```

i.e. check for the "old" env variable first then check for the "new" one then fall back to the default provider

Reply:
My expected workflow is using the default RPC URL, and only specifying a `HARDHAT_` RPC URL if I experience problems with that (and the default RPC at `@api3/chains` should be updated if the problem is persistent). Assuming that applies to most users, a breaking change is not a big deal.
I wanted to specify `HARDHAT_` because of cases like [this](https://github.com/api3dao/merkle-funder/blob/main/example.env). Without the prefix, the user may expect that the RPC URL entered here is used in the deployed funder bot. That being said, I don't like to add another duplication. Maybe `API3_CHAINS_HTTP_RPC_URL_` instead?

Reply:
works for me

Reply:
I don't think there's much reason to break this now and release a v5.0.0 so I've just added a TODO to the Hardhat version

Reply:
Makes sense